src/domain/user.js

```diff
@@ -132,7 +138,11 @@ export default class User {
     return User._findByUnique('email', email)
   }
 
-  static async findById(id) {
+  static async findById(id, role) {
+    const isTeacher = role === 'TEACHER'
+    if (isTeacher) {
+      return _findByUniqueAsATeacher(id, role)
+    }
```

Comment:
I won't stop this from merging but a better solution here would be some optional parameters to include the relevant relations / fields. Something like `includeProfile = false, includeNotes = false`